ablateLibrary/domain/fieldDescription.cpp

```diff
@@ -0,0 +1,119 @@
+#include "fieldDescription.hpp"
+#include <map>
+#include <regex>
+#include <utilities/mpiError.hpp>
+#include <utilities/petscError.hpp>
+#include <utilities/petscOptions.hpp>
+
+ablate::domain::FieldDescription::FieldDescription(std::string nameIn, std::string prefixIn, std::vector<std::string> componentsIn, ablate::domain::FieldLocation location,
+                                                   ablate::domain::FieldType type, std::shared_ptr<domain::Region> region, std::shared_ptr<parameters::Parameters> optionsIn)
+    : name(nameIn),
+      prefix(prefixIn.empty() ? name + "_" : prefixIn + "_"),
+      components(componentsIn.empty() ? std::vector<std::string>{"_"} : componentsIn),
+      location(location),
+      type(type),
+      region(region) {
+    if (optionsIn) {
+        PetscOptionsCreate(&options);
+        optionsIn->Fill(options);
+    }
+}
+
+PetscObject ablate::domain::FieldDescription::CreatePetscField(DM dm) const {
+    switch (type) {
+        case FieldType::FEM: {
+            // determine if it a simplex element and the number of dimensions
+            DMPolytopeType ct;
+            PetscInt cStart;
+            DMPlexGetHeightStratum(dm, 0, &cStart, NULL) >> checkError;
+            DMPlexGetCellType(dm, cStart, &ct) >> checkError;
+            PetscInt simplex = DMPolytopeTypeGetNumVertices(ct) == DMPolytopeTypeGetDim(ct) + 1 ? PETSC_TRUE : PETSC_FALSE;
```

Comment:
You can just use DMPlexIsSimplex() here I think.

Reply:
Thanks!  I updated the code.